lib/jekyll/filters.rb

```diff
@@ -225,6 +226,26 @@ def where(input, property, value)
       input.select { |object| item_property(object, property).to_s == value.to_s }
     end
 
+    # Filters an array of objects against an expression
+    #
+    # input - the object array
+    # variable - the variable to assign each item to in the expression
+    # expression - a Liquid comparison expression passed in as a string
+    #
+    # Returns the filtered array of objects
+    def where_exp(input, variable, expression)
+      return input unless input.is_a?(Enumerable)
+      input = input.values if input.is_a?(Hash)
```

Comment:
My motivation was to follow the existing semantics of `where` over hashes as in https://github.com/jekyll/jekyll/blob/b06af5a44f47ec6cf5f203e8eb318868eac0ae86/lib/jekyll/filters.rb#L224